### PR TITLE
[Merged by Bors] - feat(Algebra/Group/Pi/Lemmas): add Pi.mulSingle_pow

### PR DESCRIPTION
**Pull request by AntoineChambert-Loir**
add two lemmas of the form `Pi.mulSingle_pow`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Pi/Lemmas.lean [hunk 1]
```diff
@@ -248,6 +248,15 @@ theorem Pi.mulSingle_div [∀ i, Group <| f i] (i : I) (x y : f i) :
     mulSingle i (x / y) = mulSingle i x / mulSingle i y :=
   (MonoidHom.mulSingle f i).map_div x y
 
+@[to_additive]
+theorem Pi.mulSingle_pow [∀ i, Monoid (f i)] (i : I) (x : f i) (n : ℕ) :
+    mulSingle i (x ^ n) = (mulSingle i x) ^ n :=
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/Pi/Lemmas.lean:253]:
```suggestion
    mulSingle i (x ^ n) = mulSingle i x ^ n :=
```

File: Mathlib/Algebra/Group/Pi/Lemmas.lean [hunk 2]
```diff
@@ -248,6 +248,15 @@ theorem Pi.mulSingle_div [∀ i, Group <| f i] (i : I) (x y : f i) :
...
+  (MonoidHom.mulSingle f i).map_pow x n
+
+@[to_additive]
+theorem Pi.mulSingle_zpow [∀ i, Group (f i)] (i : I) (x : f i) (n : ℤ) :
+    Pi.mulSingle i (x ^ n) = (Pi.mulSingle i x) ^ n :=
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/Pi/Lemmas.lean:258]:
```suggestion
    mulSingle i (x ^ n) = mulSingle i x ^ n :=
```

**Review by b-mehta — CHANGES_REQUESTED** [review]

File: Mathlib/Algebra/Group/Pi/Lemmas.lean [hunk 3]
```diff
@@ -248,6 +248,15 @@ theorem Pi.mulSingle_div [∀ i, Group <| f i] (i : I) (x y : f i) :
...
+  (MonoidHom.mulSingle f i).map_zpow x not
```
Comment (b-mehta) [inline Mathlib/Algebra/Group/Pi/Lemmas.lean:259]:
```suggestion
  (MonoidHom.mulSingle f i).map_zpow x n
```
I guess this is what you intended to say?

Comment (AntoineChambert-Loir) [reply Mathlib/Algebra/Group/Pi/Lemmas.lean]: yes, last minute typo…

**Comment by b-mehta** [comment]
Thanks!

bors r+

**Pull request closed without merging**